src/app/CASESessionManager.h

```diff
@@ -74,18 +74,43 @@ class CASESessionManager : public Dnssd::ResolverDelegate
     virtual ~CASESessionManager() { mDNSResolver.Shutdown(); }
 
     /**
-     * Find an existing session for the given node ID, or trigger a new session request.
+     * Retain an existing session for the given node ID, or trigger a new session request.
      * The caller can optionally provide `onConnection` and `onFailure` callback objects. If provided,
      * these will be used to inform the caller about successful or failed connection establishment.
      * If the connection is already established, the `onConnection` callback will be immediately called.
+     *
+     * The reference count of the session will be incremented and shall be released by ReleaseSession when
+     * no longer used.
+     *
      */
-    CHIP_ERROR FindOrEstablishSession(PeerId peerId, Callback::Callback<OnDeviceConnected> * onConnection,
-                                      Callback::Callback<OnDeviceConnectionFailure> * onFailure);
+    CHIP_ERROR RetainSession(PeerId peerId, Callback::Callback<OnDeviceConnected> * onConnection,
```

Comment:
How does this interact with the spec-defined LRU eviction?  Is the (planned) setup that sessions that have a nonzero refcount can still be evicted via LRU?  But that a refcount goes to zero eagerly evicts?  Or something else?

Reply:
In fact, these sessions aren't _those_ sessions.  Actually, they aren't sessions at all.

CASESessionManager manages a pool of OperationalDeviceProxy objects, each of which can have zero or one CASE sessions, where the OperationalDeviceProxy lifetime is longer / different than the lifetime of an actual session.  Also, CASESessionManager is not related to or extending SessionManager.

I think we are just suffering from some suboptimal naming.

Reply:
> How does this interact with the spec-defined LRU eviction?

Sorry, I should have answered this more specifically.  That would happen in SessionManager, and wouldn't have any dependence on any of the code here.  Although, CASESessionManager could be notified of eviction.

But again, I think this is all just a situation of naming that could be better.